### ISSUE TITLE
RNTester on iOS with USE_FABRIC=1 no longer builds

### ISSUE THREAD

**Issue by terrysahaidak** [facebook/react-native#31997]
## Description

I'm not able to build the RNTester app with fabric enabled on the simulator. Last time a month-ish ago I was able to do so, after fetching latest changes it no longer builds, throws errors attached below:

<details>
<img src="https://user-images.githubusercontent.com/7809008/128876613-f820ef7b-b036-4168-85d5-09ec385ec830.png" />
</details>

Lem me know if I can provide more information to help figure it out.

From the error, I can see that some of the files are treated as Cpp files instead of Obj-Cpp.

## React Native version:

    OS: macOS 10.15.7
    CPU: (12) x64 Intel(R) Core(TM) i7-8750H CPU @ 2.20GHz
    Memory: 838.98 MB / 16.00 GB
    Shell: 5.7.1 - /bin/zsh
  Binaries:
    Node: 14.17.1 - ~/.nvm/versions/node/v14.17.1/bin/node
    Yarn: 1.5.1 - ~/.yarn/bin/yarn
    npm: 6.14.13 - ~/.nvm/versions/node/v14.17.1/bin/npm
    Watchman: 4.9.0 - /usr/local/bin/watchman
  Managers:
    CocoaPods: 1.10.1 - /Users/terry/.rvm/gems/ruby-2.4.6/bin/pod
  SDKs:
    iOS SDK:
      Platforms: iOS 14.4, DriverKit 20.2, macOS 11.1, tvOS 14.3, watchOS 7.2
    Android SDK:
      API Levels: 23, 26, 28, 29, 30, 31
      Build Tools: 27.0.3, 28.0.3, 29.0.2, 30.0.2, 31.0.0
      Android NDK: 21.3.6528147
  IDEs:
    Android Studio: 3.5 AI-191.8026.42.35.5900203
    Xcode: 12.4/12D4e - /usr/bin/xcodebuild
  Languages:
    Java: 1.8.0_252 - /usr/bin/javac
  npmPackages:
    @react-native-community/cli: Not Found
    react: Not Found
    react-native: Not Found
    react-native-macos: Not Found
  npmGlobalPackages:
    *react-native*: Not Found

## Steps To Reproduce
Provide a detailed list of steps that reproduce the issue.

1. Clone react-native repo
2. run `yarn && cd packages/rn-tester && yarn && cd ios && USE_FABRIC=1 pod install`
3. open in xcode
4. build on simulator

## Expected Results

It should builds with no errors.

## Snack, code example, screenshot, or link to a repository:

Clean recent repo clone

**Comment by chrisglein** [comment]
Looks like your environment command didn't pick up the RN version. Can you try that again? (you might need to run it from a different location).

Also, can you provide the text for your build errors? (hard to get from the screenshot)

**Comment by terrysahaidak** [comment]
I'm running the RNTester app from the latest main branch.

I've just pulled latest changes and now I'm getting way different errors :) 
Not sure how it works.

Each build - different results.

But one thing I've just realized - I had to run `build` script of `react-native-codegen`. Probably I runs with "use_react_native_codegen" ruby script but it helped me to get different results at least.

The very last time I had removed `Pods` and `Podfile.lock`, run `build` of `react-native-codegen` and installed pods with `USE_FABRIC=1`, cleared build folder and now it builds without any errors.

So for anyone having any problems with building RNTester - make sure to execute codegen before installation of the Pods.

Not sure it'll stay like that in the future.
